docs/configuration/project-references.md

```diff
@@ -0,0 +1,147 @@
+
+{% endhint %}
+
+As with project references in general, there are a few extra pieces of configuration to add, though.
+
+[pr]: http://www.typescriptlang.org/docs/handbook/project-references.html#composite
+
+## Shared project-wide configuration
+
+First, as with project references in general, you should define your `glint` config in a shared compiler options `tsconfig.json` at the root. For example, you might have something like this as a `tsconfig.shared.json`:
+
+```jsonc
+{
+  "compilerOptions": {
+    // ...
+  },
+  "glint": {
+    "environment": "ember-template-imports"
+  }
+}
+```
+
+Then each project within can use the `extends` key to reuse both TypeScript and Glint configuration. For example, if your project has sub-projects all located in `packages/<project>`, a sub-project might have a `tsconfig.json` file like this:
+
+```jsonc
+{
+  "extends": "../../tsconfig.shared.json",
+  "compilerOptions": {
+    "composite": true
+  }
+}
+```
+
+## Per-project configuration
+
+To work with Glint, each sub-project needs to include the Glint environment imports it uses. (See [Ember: Installation][ei] and [GlimmerX: Imports][gi] for details.) For example, a project using `ember-template-imports` should have this import visible somewhere:
+
+```ts
+import '@glint/environment-ember-template-imports';
+```
+
+[ei]: ../ember/installation.md
+[gi]: ../glimmerx/imports.md
+
+The easiest way to do this is to add it to a shared file all projects can reference using the `include` key in each sub-project's `tsconfig.json`.
+If you do not already have something filling this role, you should introduce a `local-types` directory (named however you like) in the root of your project, with an `index.d.ts` in it.
+
+{% hint style="info" %}
+
+If you are using `ember-cli-typescript`, you can do this in the `types/index.d.ts` file created automatically for you by the addon. Just make sure you add that directory to the `include` field in all your sub-project `tsconfig.json` files.
+
+{% endhint %}
+
+The resulting project structure might look something like this, given sub-projects in `packages` which have their TypeScript source files in a `src` directory:
+
+```
+.
+├── package.json
+├── packages
+│   ├── a
+│   │   ├── local-types
+│   │   │   └── index.d.ts
+│   │   ├── package.json
+│   │   ├── src
+│   │   └── tsconfig.json
+│   └── b
+│       ├── package.json
+│       ├── src
+│       └── tsconfig.json
+├── tsconfig.json
+├── tsconfig.shared.json
+└── local-types
+    └── index.d.ts
+```
+
+With this structure, the sub-project configurations, like `packages/a/tsconfig.json`, can be very minimal. The only things they need to do are:
+
+- extend the root options config
+- include the `types` directory as well as their own files
+
+```jsonc
+{
+  "extends": "../../tsconfig.shared.json",
+  "compilerOptions": {
+    "composite": true
+  },
+  "include": ["../../local-types/**/*.ts", "./src/**/*.ts"]
```

Comment:
It's been a long time, but I _think_ the original reason we had to do `**/*.ts` in these was something to do with watcher issues when we were using `tsc` to do compilation in `ember-cli-typescript` (way back in 1.x). Given that, this could probably be just `["../../local-types", "src"]`.

Reply:
Updated in 725e32e!